parquet-column/src/main/java/org/apache/parquet/column/page/PageWriter.java

```diff
@@ -39,9 +38,25 @@
    * @param dlEncoding definition level encoding
    * @param valuesEncoding values encoding
    * @throws IOException if there is an exception while writing page data
+   * @deprecated will be removed in 2.0.0. This method does not support writing column indexes; Use
+   *             {@link #writePage(BytesInput, int, int, Statistics, Encoding, Encoding, Encoding)} instead
    */
+  @Deprecated
```

Comment:
I don't think this requires deprecation. It's part of the column API that is internal.

Reply:
Unfortunately, it is not documented anywhere and we do know that at least two different projects use it directly (Spark and Hive for vectorization).
I think, it is better to be on the safe side and handle every java-public API members as Public so we won't break any API consumer.